spring-batch-infrastructure/src/test/java/org/springframework/batch/item/ExecutionContextTests.java

```diff
@@ -88,13 +88,13 @@ void testNotDirtyWithDuplicate() {
 	}
 
 	@Test
-	void testNotDirtyWithRemoveMissing() {
+	void testDirtyWithRemoveMissing() {
 		context.putString("1", "test");
 		assertTrue(context.isDirty());
 		context.putString("1", null); // remove an item that was present
 		assertTrue(context.isDirty());
 		context.putString("1", null); // remove a non-existent item
-		assertFalse(context.isDirty());
+		assertTrue(context.isDirty());
```

Comment:
By changing this line you change the meaning of the test. I think it would be better to clear the dirty flag before setting it to null for the second time

Reply:
actually testing that it stays dirty when inserting null twice could then be done in its own test or testDirtyWithDuplicate

Reply:
I made some adjustments to the test based on the feedback. I added a step to clear the dirty flag before trying to remove an already non-existent item. This way, the test clearly checks that the dirty state doesn’t change in that situation. I think it makes the intent of the test more explicit and separates the different scenarios, making it easier to understand.